From 0ffd58b223e24779420130522ea8ee359505f493 Mon Sep 17 00:00:00 2001
From: Gang He <ghe@suse.com>
Date: Mon, 4 Sep 2017 14:08:59 +0800
Subject: [PATCH] fsck.ocfs2: fix compile error when glibc upgrade

When glibc upgrade to glibc-2.26.90-14, there was a compile error in fsck.ocfs2,
the compile error messages like,

In file included from /usr/include/string.h:431:0,
                 from ../include/ocfs2/ocfs2.h:41,
                 from pass4.c:32:
include/strings.h:37:1: error: unknown type name ‘errcode_t’; did you mean ‘mode_t’?
 errcode_t o2fsck_strings_insert(o2fsck_strings *strings, char *string,
 ^~~~~~~~~
 mode_t

Patch-Source: https://github.com/markfasheh/ocfs2-tools/commit/0ffd58b223e24779420130522ea8ee359505f493
---
 fsck.ocfs2/Makefile                                | 2 +-
 fsck.ocfs2/include/{strings.h => o2fsck_strings.h} | 0
 fsck.ocfs2/pass2.c                                 | 2 +-
 fsck.ocfs2/pass5.c                                 | 2 +-
 fsck.ocfs2/strings.c                               | 2 +-
 5 files changed, 4 insertions(+), 4 deletions(-)
 rename fsck.ocfs2/include/{strings.h => o2fsck_strings.h} (100%)

diff --git a/fsck.ocfs2/Makefile b/fsck.ocfs2/Makefile
index 051ed742..baf1994d 100644
--- a/fsck.ocfs2/Makefile
+++ b/fsck.ocfs2/Makefile
@@ -64,7 +64,7 @@ HFILES = 	include/fsck.h		\
 		include/problem.h	\
 		include/refcount.h	\
 		include/slot_recovery.h	\
-		include/strings.h	\
+		include/o2fsck_strings.h	\
 		include/util.h
 
 
diff --git a/fsck.ocfs2/include/strings.h b/fsck.ocfs2/include/o2fsck_strings.h
similarity index 100%
rename from fsck.ocfs2/include/strings.h
rename to fsck.ocfs2/include/o2fsck_strings.h
diff --git a/fsck.ocfs2/pass2.c b/fsck.ocfs2/pass2.c
index 181febb9..a48a50b5 100644
--- a/fsck.ocfs2/pass2.c
+++ b/fsck.ocfs2/pass2.c
@@ -43,7 +43,7 @@
 #include "fsck.h"
 #include "pass2.h"
 #include "problem.h"
-#include "strings.h"
+#include "o2fsck_strings.h"
 #include "util.h"
 
 static const char *whoami = "pass2";
diff --git a/fsck.ocfs2/pass5.c b/fsck.ocfs2/pass5.c
index 5c2d899f..bfad1b72 100644
--- a/fsck.ocfs2/pass5.c
+++ b/fsck.ocfs2/pass5.c
@@ -40,7 +40,7 @@
 #include "fsck.h"
 #include "pass5.h"
 #include "problem.h"
-#include "strings.h"
+#include "o2fsck_strings.h"
 #include "util.h"
 
 static const char *whoami = "pass5";
diff --git a/fsck.ocfs2/strings.c b/fsck.ocfs2/strings.c
index 4ad37822..8a7a8d9d 100644
--- a/fsck.ocfs2/strings.c
+++ b/fsck.ocfs2/strings.c
@@ -31,7 +31,7 @@
 #include "ocfs2/ocfs2.h"
 
 #include "fsck.h"
-#include "strings.h"
+#include "o2fsck_strings.h"
 #include "util.h"
 
 struct string_entry {
